src/ChangeSet.ts

```diff
@@ -188,6 +200,15 @@ export class ChangeSet {
       }
     }
   }
+  /* Checks if the `nextChange` immediately follows the `currentChange` (i.e., the start position of `nextChange` matches 
+ the end position of `currentChange`)and ensures both are text changes before merging. */
+  mergeTextChanges(currentChange: TrackedChange, nextChange: TrackedChange): boolean {
+    return (
+      currentChange.to === nextChange.from &&
```

Comment:
we need also to check `currentChange.dataTracked.status === nextChange.dataTracked.status`

Reply:
Thanks, @asouqi  Yes, I just noticed the issue when approving/rejecting the suggestion. I'm working on fixing that.